cmd/httpapi/main.go

```diff
@@ -25,22 +32,74 @@ func readCryptoKey(privateKeyPath string, publicKeyPath string) (privateKeyBytes
 func main() {
     privateKeyPath := flag.String("privateKey", "app.rsa", "file path")
     publicKeyPath := flag.String("publicKey", "app.rsa.pub", "file path")
+    articleDataPath := flag.String("articleDataPath", "data.txt", "file path")
     flag.Parse()
 
     privateKeyBytes, publicKeyBytes, err := readCryptoKey(*privateKeyPath, *publicKeyPath)
     if err != nil {
         panic(err)
     }
 
-    a, err := auth.NewJwtConfig(privateKeyBytes, publicKeyBytes, 100*time.Minute)
+    jwtAuth, err := auth.NewJwtConfig(privateKeyBytes, publicKeyBytes, 100*time.Minute)
     if err != nil {
         panic(err)
     }
 
-    //unarXivUsecases := dummyUsecases.DummyUsecases{}
-    unarXivUsecases := smartUsecases.SmartUsecases{AccountStorage: memory.NewAccountsRepo(), Auth: a}
+    authUsecases := auth.NewUsecases(memory.NewAccountsRepo(), jwtAuth)
+    articleRepo := memory.NewArticleRepo()
+    historyRepo := memory.NewHistoryRepo()
+    artSubsRepo := memory.NewArticleSubscriptionRepo()
+    searchSubsRepo := memory.NewSearchSubscriptionRepo()
+    updatesRepo := repository.NewUpdatesRepoThroughQueries(articleRepo, artSubsRepo, searchSubsRepo, historyRepo)
 
-    httpApi := httpapi.New(&unarXivUsecases)
+    if articleDataPath != nil {
```

Comment:
Oh no, please spare me reading all these nested spaghetti.. Refactor immediately! No more than 1 (1,5 max) level of nesting should be ever approved.

Reply:
fixed

Reply:
oops, wait a minute please, gotta fix merge conflicts

Reply:
done